sklearnex/ensemble/forest.py

```diff
@@ -13,6 +14,941 @@
+                    "%r" %
+                    self.max_leaf_nodes)
+            if self.max_leaf_nodes < 2:
+                raise ValueError(
+                    ("max_leaf_nodes {0} must be either None "
+                     "or larger than 1").format(
+                        self.max_leaf_nodes))
+        if isinstance(self.max_bins, numbers.Integral):
+            if not 2 <= self.max_bins:
+                raise ValueError("max_bins must be at least 2, got %s"
+                                 % self.max_bins)
+        else:
+            raise ValueError("max_bins must be integral number but was "
+                             "%r" % self.max_bins)
+        if isinstance(self.min_bin_size, numbers.Integral):
+            if not 1 <= self.min_bin_size:
+                raise ValueError("min_bin_size must be at least 1, got %s"
+                                 % self.min_bin_size)
+        else:
+            raise ValueError("min_bin_size must be integral number but was "
+                             "%r" % self.min_bin_size)
+
+    def check_sample_weight(sample_weight, X, dtype=None):
+        n_samples = _num_samples(X)
+
+        if dtype is not None and dtype not in [np.float32, np.float64]:
+            dtype = np.float64
+
+        if sample_weight is None:
+            sample_weight = np.ones(n_samples, dtype=dtype)
+        elif isinstance(sample_weight, numbers.Number):
+            sample_weight = np.full(n_samples, sample_weight, dtype=dtype)
+        else:
+            if dtype is None:
+                dtype = [np.float64, np.float32]
+            sample_weight = check_array(
+                sample_weight,
+                accept_sparse=False,
+                ensure_2d=False,
+                dtype=dtype,
+                order="C")
+            if sample_weight.ndim != 1:
+                raise ValueError("Sample weights must be 1D array or scalar")
+
+            if sample_weight.shape != (n_samples,):
+                raise ValueError("sample_weight.shape == {}, expected {}!"
+                                 .format(sample_weight.shape, (n_samples,)))
+        return sample_weight
+
+    def _onedal_fit_data_supported(self, X, y, sample_weight):
+        try:
+            X_, y_ = _check_X_y(
+                X, y, dtype=[np.float64, np.float32],
+                 force_all_finite=True, accept_sparse=False)
+            y_ = np.asarray(y_)
+            if y_.ndim == 1:
+                # reshape is necessary to preserve the data contiguity against vs
+                # [:, np.newaxis] that does not.
+                y_ = np.reshape(y_, (-1, 1))
+            n_outputs_ = y_.shape[1]
+            if n_outputs_ != 1:
+                raise ValueError(
+                    "Number of outputs ({n_outputs_}) is not 1."
+                )
+            if sample_weight is not None:
+                sample_weight_ = self.check_sample_weight(sample_weight, X)

```

Comment:
If sample_weight is None, _onedal_fit_data_supported drops in exception with resulting `self._fit_data_support = False`
```suggestion
            if sample_weight is not None:
                sample_weight_ = self.check_sample_weight(sample_weight, X)
            else:
                sample_weight_ = None
```

Reply:
This changes were reverted.